src/d3d12/nvapi_d3d12_device.cpp

```diff
@@ -59,11 +86,27 @@ namespace dxvk {
     }
 
     bool NvapiD3d12Device::LaunchCubinShader(ID3D12GraphicsCommandList* commandList, NVDX_ObjectHandle pShader, NvU32 blockX, NvU32 blockY, NvU32 blockZ, const void* params, NvU32 paramSize) {
-        auto commandListExt = GetCommandListExt(commandList);
+        uint32_t interfaceVersion = 0;
+        auto commandListExt = GetCommandListExt(commandList, &interfaceVersion);
         if (commandListExt == nullptr)
             return false;
 
-        return SUCCEEDED(commandListExt->LaunchCubinShader(reinterpret_cast<D3D12_CUBIN_DATA_HANDLE*>(pShader), blockX, blockY, blockZ, params, paramSize));
+        uint32_t smem = 0;
+        {
+            // Hop into a scope here to not take the lock when calling into
+            // vkd3d-proton
+            std::scoped_lock lock(m_CubinSmemMutex);
+            auto it = m_cubinSmemMap.find(pShader);
+            if (it != m_cubinSmemMap.end())
+                smem = it->second;
+            else
+                log::write("Failed to find CuBIN in m_cubinSmemMap, defaulting to 0");
+        }
+
+        if (interfaceVersion >= 1)
+            return SUCCEEDED(commandListExt->LaunchCubinShaderEx(reinterpret_cast<D3D12_CUBIN_DATA_HANDLE*>(pShader), blockX, blockY, blockZ, smem, params, paramSize, nullptr, 0));
+        else
```

Comment:
I think finding a non-zero smem size in the map when interface version is `< 1` also warrants for a log message warning about this.

Reply:
Agreed, will add in next update.